src/mpi/init/mpir_init.c

```diff
@@ -111,6 +111,16 @@ int MPII_Init_thread(int *argc, char ***argv, int user_required, int *provided,
     if (!is_world_model) {
         *p_session_ptr = (MPIR_Session *) MPIR_Handle_obj_alloc(&MPIR_Session_mem);
         MPIR_ERR_CHKHANDLEMEM(*p_session_ptr);
+
+        (*p_session_ptr)->errhandler = NULL;
+        /* FIXME: actually do something with session thread_level */
+        (*p_session_ptr)->thread_level = user_required;
+
+        {
+            int thr_err;
+            MPID_Thread_mutex_create(&(*p_session_ptr)->mutex, &thr_err);
+            MPIR_Assert(thr_err == 0);
```

Comment:
where is this mutex destroyed? it's not actually used in this commit, either.

Reply:
I see the destroy code was added in a later commit. Not using it in the commit it is added is fine, to be clear.

Reply:
Moved the destroy of mutex to the same commit.